test/server/controller/plugin.test.ts

```diff
@@ -33,102 +34,180 @@ describe("Controllers Test", function () {
 
     afterEach(function () {
       jest.resetAllMocks();
+      jest.resetModules();
     });
 
     it("should write a trade to db and return an id", async function () {
       const result: {
-        [rows: string]: [{ [id: string]: "" }];
+        rows: Array<{ id: string }>;
       } = await controller.write(trade);
       expect(result.rows[0].id).toEqual("abc");
-      expect(
-        pg.Client.prototype.query
-      ).toHaveBeenCalledWith(
-        `INSERT INTO trades (id, ticker, company_name, reference_number, unit_price, quantity, total_cost, trade_type, note, created_at, trade_status) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11) RETURNING id`,
-        ["abc", "ABC", "Test", "", 0, 0, 0, "long", "", "10/10/2020", true]
-      );
+      expect(PostgresGateway.prototype.write).toHaveBeenCalledWith({
+        company_name: "Test",
+        created_at: "10/10/2020",
+        id: "abc",
+        note: "",
+        quantity: 0,
+        reference_number: "",
+        ticker: "ABC",
+        total_cost: 0,
+        trade_status: true,
+        trade_type: "long",
+        unit_price: 0,
+      });
     });
 
-    it("should throw an error if query is faulty", async function () {
-      pg.Client.prototype.query = jest
-        .fn()
-        .mockRejectedValue(new Error("Test Error"));
-      expect(typeof controller.write(trade)).toEqual("object");
+    it("should log the error with Winston logger if the query fails", async function (done) {
+      try {
+        PostgresGateway.prototype.write = jest
+          .fn()
+          .mockRejectedValue(new Error("Test Error"));
+
+        await controller.readByDate(trade.created_at);
+      } catch (err) {
+        expect(err.message).toEqual("Fake Error");
+        expect(logger).toHaveBeenCalledTimes(1);
```

Comment:
**Comment**: As an add-on to your newly introduced behavior, you should also expect that the logger was called with your `Test Error`. This ensures that the stack trace will be correctly preserved.

Reply:
Added new expect clause to cover this.